### DISCUSSION TITLE
Issues to look out for?

### DISCUSSION THREAD

**Discussion by jawadd1** [TagStudioDev/TagStudio#303] [General]
### Use case scenario:
Using TagStudio to tag all my files, and even **folders**. Or at least folders should be embedded with the tags of the files within them. (I guess we can consider that a suggestion).
Then, all my files and folders will be synced across my devices and clouds (Android, Windows, Google Drive and OneDrive). I'll probably use all/some/one of SyncThing, FolderSync, and both clouds' desktop apps.

Despite TagStudio being the perfect fit for this, as described by your outstanding video, there are;

### Worries
1. Tagging files that are in use by applications might cause dysfunction.
**e.g.** tagging an that's in my WhatsApp folder might result in that image not appearing in its respective chat.

2. Tagging files might change the file's date.

3. Tagging files would cause duplicates in backups and synchronisation processes.
- Say you're syncing your files with another device, or the cloud. After tagging, your sync handler/cloud won't recognise the file as the same, so it'll back up the file as a new one.
-- This would be okay if it would **replace** the old file with the new one, but not very much if it uploads a copy.

4. Clouds would strip the tags/metadata when uploading the file.
- You upload the file, then download it to find that it's not tagged at all.

I would also like to volunteer to test whatever version of the program you want.

Cannot thank you enough for the tremendous work.

**Comment by CyanVoxel** [comment]
TagStudio stores all tag and metadata information for files inside of a central per-library database, separate from the files themselves. That way there isn't any issues with modifying or accessing files, among other benefits. This data is currently stored inside the `ts_library.json` file inside the `.TagStudio` folder that's created when creating a new library. So while this opens up some other cases that need to be considered for TagStudio's sake, it prevents each of the concerns you've mentioned :)

And we're always looking for people to help test the program! Whether that be the [official releases](https://github.com/TagStudioDev/TagStudio/releases) or the development branches via a live [development environment](https://github.com/TagStudioDev/TagStudio/blob/main/CONTRIBUTING.md)
